pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/SanitizationTransformer.java

```diff
@@ -51,24 +63,35 @@ public SanitizationTransformer(Schema schema) {
   @Override
   public GenericRow transform(GenericRow record) {
     for (Map.Entry<String, Integer> entry : _stringColumnMaxLengthMap.entrySet()) {
-      String stringColumn = entry.getKey();
-      int maxLength = entry.getValue();
-      Object value = record.getValue(stringColumn);
-      if (value instanceof String) {
-        // Single-valued column
-        String stringValue = (String) value;
-        String sanitizedValue = StringUtil.sanitizeStringValue(stringValue, maxLength);
-        // NOTE: reference comparison
-        //noinspection StringEquality
-        if (sanitizedValue != stringValue) {
-          record.putValue(stringColumn, sanitizedValue);
+      try {
+        String stringColumn = entry.getKey();
+        int maxLength = entry.getValue();
+        Object value = record.getValue(stringColumn);
+        if (value instanceof String) {
+          // Single-valued column
+          String stringValue = (String) value;
+          String sanitizedValue = StringUtil.sanitizeStringValue(stringValue, maxLength);
+          // NOTE: reference comparison
+          //noinspection StringEquality
+          if (sanitizedValue != stringValue) {
+            record.putValue(stringColumn, sanitizedValue);
+          }
+        } else {
+          // Multi-valued column
+          Object[] values = (Object[]) value;
+          int numValues = values.length;
+          for (int i = 0; i < numValues; i++) {
+            values[i] = StringUtil.sanitizeStringValue(values[i].toString(), maxLength);
+          }
         }
-      } else {
-        // Multi-valued column
-        Object[] values = (Object[]) value;
-        int numValues = values.length;
-        for (int i = 0; i < numValues; i++) {
-          values[i] = StringUtil.sanitizeStringValue(values[i].toString(), maxLength);
+      } catch (Exception e) {
+        if (!_continueOnError) {
+          throw new RuntimeException("Caught exception while sanitizing value for column: " + entry.getKey(), e);
+        } else {
+          LOGGER.debug("Caught exception while sanitizing value for column: {}", entry.getKey(), e);
+          //TODO: should put NULL here instead of string `null` and use NullValueTransformer for appropriate value
+          record.putValue(entry.getKey(), "null");
+          record.putValue(GenericRow.INCOMPLETE_RECORD_KEY, true);
```

Comment:
seems other transformers were already adding this value to the GenericRow, but until this PR, it would've been simply ignored from final segment?